### PR TITLE
Add triage form

### PR DESCRIPTION
**Pull request by misaka**
Adds triage notes to patient triage page.

Note the change to the routes table (commit: [Nest triage route under sessions/patients](https://github.com/nhsuk/record-childrens-vaccinations/commit/ec6322c5d56efaaeaadd45eaf4369637fdc41571)). This change may be controversial and worth a discussion. If accepted it'll mean changes to the vaccinations routes (probably).

image

### CODE REVIEW

File: app/views/triage/_patient_triage_card.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<div class="nhsuk-card">
+  <div class="nhsuk-card__content">
+    <h2 class="nhsuk-card__heading nhsuk-heading-m">
+      Triage
+    </h2>
+
+    <%= form_for [session, triage], builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
```
Comment (tvararu) [inline app/views/triage/_patient_triage_card.html.erb]: Do you need to specify the builder? It should be the default, I didn't have to on my branch...

File: app/controllers/triage_controller.rb [hunk 2]
```diff
@@ -21,6 +21,12 @@ def index
   def show
   end
 
+  def create
+    @triage = Triage.new(campaign: @session.campaign)
+    @triage.update!(triage_params)
+    redirect_to session_triage_index_path(@session)
```
Comment (tvararu) [inline app/controllers/triage_controller.rb]: Should we handle potential errors here at this stage, or add it in later?

File: app/views/triage/_patient_triage_card.html.erb [hunk 3]
```diff
@@ -5,6 +5,7 @@
     </h2>
 
     <%= form_for [session, triage], builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
+      <%= hidden_field(:triage, :patient_id) %>
```
Comment (tvararu) [inline app/views/triage/_patient_triage_card.html.erb]: Does `f.hidden_field` work, is it preferable?

**Review by tvararu — APPROVED** [review]

File: app/views/triage/_patient_triage_card.html.erb (hunk 3 above)
Comment (misaka) [reply app/views/triage/_patient_triage_card.html.erb]: I got rid of this in the last change to accommodate update, found a way around it.

File: app/views/triage/_patient_triage_card.html.erb (hunk 1 above)
Comment (misaka) [reply app/views/triage/_patient_triage_card.html.erb]: Nope, you're right I don't. Removed.

File: app/controllers/triage_controller.rb (hunk 2 above)
Comment (misaka) [reply app/controllers/triage_controller.rb]: You're right. Have you got an example of how to do this to hand?

File: app/controllers/triage_controller.rb [hunk 4]
```diff
@@ -24,14 +24,20 @@ def show
 
   def create
     @triage = Triage.new(campaign: @session.campaign, patient: @patient)
-    @triage.update!(triage_params)
-    redirect_to triage_session_path(@session)
+    if @triage.update(triage_params)
+      redirect_to triage_session_path(@session)
+    else
+      render :show
```
Comment (tvararu) [inline app/controllers/triage_controller.rb:33]: We should add `status: :unprocessable_entity` here

File: app/controllers/triage_controller.rb [hunk 5]
```diff
@@ -24,14 +24,20 @@ def show
...
   end
 
   def update
     @triage = @patient.triage_for_campaign(@session.campaign)
-    @triage.update!(triage_params)
-    redirect_to triage_session_path(@session)
+    if @triage.update(triage_params)
+      redirect_to triage_session_path(@session)
+    else
+      render :show
```
Comment (tvararu) [inline app/controllers/triage_controller.rb:42]: And here

File: app/controllers/triage_controller.rb (hunk 2 above)
Comment (misaka) [reply app/controllers/triage_controller.rb]: Done.

**Review by tvararu — APPROVED** [review]

**Pull request merged**